tests/components/device_tracker/test_ritassist.py

```diff
@@ -0,0 +1,60 @@
+"""The tests for the RitAssist device tracker."""
+import unittest
+import requests_mock
+from tests.common import load_fixture, get_test_home_assistant
+
+from homeassistant.components.device_tracker.ritassist import (
+    RitAssistDeviceScanner)
+
+
+class TestRitAssistSetup(unittest.TestCase):
+    """Test the RitAssist device tracker."""
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.devices = []
+        self.hass = get_test_home_assistant()
+        self.config = {
+            'device_tracker': {
+                'platform': 'ritassist',
+                'client_id': 'client_id',
+                'client_secret': 'client_secret',
+                'username': 'username',
+                'password': 'passwword',
+                'include': [
+                    'AA-123-A'
+                ]
+            }
+        }
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @requests_mock.Mocker()
+    def test_setup(self, mock_req):
+        """Test for successfully setting up the platform."""
+        self._setup_mock_requests(mock_req)
+
+        scanner = RitAssistDeviceScanner(self.hass, self.config, None, None)
```

Comment:
Please use the home assistant core to set up the device tracker component and ritassist platform. Then use the state machine to get the state and assert that it's correct.

Use the `setup_component` function from homeassistant.setup.py.

Reply:
Using `setup_component` causes a couple of errors, mainly:

```
ERROR:homeassistant.setup:Error during setup of component device_tracker
Traceback (most recent call last):
  File "/Volumes/Projects/forks/home-assistant/homeassistant/setup.py", line 146, in _async_setup_component
    component.setup, hass, processed_config)  # type: ignore
AttributeError: module 'custom_components.device_tracker' has no attribute 'setup'
INFO:homeassistant.core:Bus:Handling <Event call_service[L]: domain=persistent_notification, service=create, service_data=title=Invalid config, message=The following components and platforms could not be set up:

 - [device-tracker](https://home-assistant.io/components/device_tracker/)
```

Reply:
Moving tests to external module. Not much to test left in this device tracker.

Reply:
Are you using a custom component?